bundles/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/internal/handler/AbstractWiFiLEDDriver.java

```diff
@@ -111,14 +111,13 @@ public void init() throws IOException {
     public abstract LEDStateDTO getLEDStateDTO() throws IOException;
 
     protected synchronized LEDState getLEDState() throws IOException {
-        try (Socket socket = new Socket(host, port)) {
+        try (Socket socket = new Socket(host, port);
+                DataOutputStream outputStream = new DataOutputStream(socket.getOutputStream());
+                DataInputStream inputStream = new DataInputStream(socket.getInputStream())) {
             logger.debug("Connected to '{}'", socket);
 
             socket.setSoTimeout(DEFAULT_SOCKET_TIMEOUT);
```

Comment:
Is this reordering safe?

Reply:
@martinvw To be honest, I do not know. Do you think it will work different? We can introduce a second try-with-resource statement at the previous place in the code.

Reply:
I think `getOutputStream` is non-blocking, so it should be ok.

Reply:
Did @wborn or @Hilbrand use this binding maybe they can quickly check (after) merging?

Reply:
I don't use this binding. But I would expect the timeout to only work on read. So the reordering should not be a problem I assume.